galasa-parent/dev.galasa.framework.api.webui/src/main/java/dev/galasa/framework/api/webui/WorklistQuery.java

```diff
@@ -0,0 +1,146 @@
+package dev.galasa.framework.api.webui;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+import dev.galasa.framework.spi.DynamicStatusStoreException;
+
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property =
+{"osgi.http.whiteboard.servlet.pattern=/webui/worklist" }, name = "Galasa Worklist microservice")
+public class WorklistQuery extends HttpServlet {
+
+	private static final long serialVersionUID = 1L;
+	  
+	@Reference
+	IFramework framework;
+	
+	Gson gson = GalasaGsonBuilder.build();
+	   
+	static List<String> runIdsInWorklist = new ArrayList<String>();
```

Comment:
cannot be static for a multithreaded environment.
an instance of this class will be created per servlet thread,  so this will break BIG time

Reply:
dont see why you need it is as a field anyway